game/src/common/waypoints.rs

```diff
@@ -221,8 +218,42 @@ impl InputWaypoints {
 
     fn update_waypoints_drawable(&mut self, ctx: &mut EventCtx) {
         let mut batch = GeomBatch::new();
-        for waypt in &self.waypoints {
-            batch.append(waypt.geom.clone());
+        let total_waypoints = self.waypoints.len();
+        for (idx, waypt) in &mut self.waypoints.iter().enumerate() {
```

Comment:
We don't need the `&mut` here. Normally you do something like `for x in &mut list` as a shorthand for `for x in list.iter_mut()`. I'm not even sure what the `&mut` does in this case...

Reply:
Removed.